### DISCUSSION TITLE
useStyles accept 1 prop only and not 2

### DISCUSSION THREAD

**Discussion by myhendry** [react-native-elements/react-native-elements#3328] [Q&A] [answered]
Hi. I followed the react-native-element documentation on how to use makeStyles with theme and props

however, I get following typescript error Expected 0-1 arguments, but got 2 

anything I did incorrectly? what should I do to successfully pass theme and props to styles? Thanks
```

import React from "react";
import { StyleSheet, Image } from "react-native";
import { makeStyles, useTheme } from "react-native-elements";

type Props = {
  size: number;
  user: any;
};

const Avatar = ({ user, ...props }: Props) => {
  const { theme } = useTheme();
  const styles = useStyles(theme, props); // <--- typescript error Expected 0-1 arguments, but got 2 

  return (
    <Image
      style={styles.image}
      source={
        user.photoURL
          ? { uri: user.photoURL }
          : require("../../assets/icon-square.png")
      }
      resizeMode="cover"
    />
  );
};

export default Avatar;

const useStyles = makeStyles((theme, props) => ({
  button: {
    width: props.size,
    height: props.size,
    borderRadius: props.size,
  },
}));
```

**Comment by arpitBhalla** [comment, marked as answer]
It'd be `const styles = useStyles(props)`

Can you share the documentation link.

**Reply by myhendry** [reply]
hi. here's the [link](https://reactnativeelements.com/docs/customization). in this case, how do i pass the theme then?

**Reply by myhendry** [reply]
hi. i copied from the documentation. how do we pass the theme then - if we only can pass props into useStyles? 

```
import React from 'react';
import { Text, View } from 'react-native';
import { makeStyles, useTheme } from 'react-native-elements';

const MyComponent = (props) => {
  const { theme } = useTheme();
  const styles = useStyles(theme,props);

  return (
    <View style={styles.container}>
      <Text style={styles.text}>Yo!</Text>
    </View>
  );
};

const useStyles = makeStyles((theme, props) => ({
  container: {
    background: theme.colors.white,
    width: props.fullWidth ? '100%' : 'auto',
  },
  text: {
    color: theme.colors.primary,
  },
}));

export default MyComponent;
```

**Reply by arpitBhalla** [reply]
You don't need to pass theme as argument, it's inside useStyles hook.
Refer here
https://github.com/react-native-elements/react-native-elements/blob/b4a729645b064ae7854eb32eeb060b8c0ac18cc2/packages/themed/src/config/makeStyles.ts#L15

**Reply by myhendry** [reply]
i see. i get what you mean now. Thanks a lot! just one other question. how do we get typings for the props in makeStyles? i get Property 'size' does not exist on type 'unknown'.

const Avatar = ({ user, ...props }: Props) => {
  const styles = useStyles(props);

  return (
    <Image
      style={styles.image}
      source={
        user.photoURL
          ? { uri: user.photoURL }
          : require("../../assets/icon-square.png")
      }
      resizeMode="cover"
    />
  );
};

export default Avatar;

const useStyles = makeStyles((theme, props) => ({
  image: {
    width: props.size, <-- Property 'size' does not exist on type 'unknown'.
    height: props.size, <-- Property 'size' does not exist on type 'unknown'.
    borderRadius: props.size, <-- Property 'size' does not exist on type 'unknown'.
  },
}));
